scripts/kytos_zabbix.py

```diff
@@ -0,0 +1,226 @@
+ 2: CACHE_DIR + '/links.json',
+ 3: CACHE_DIR + '/evcs.json',
+ 4: CACHE_DIR + '/flows.json',
+ 5: CACHE_DIR + '/flows.json',
+}
+
+help_msg = "Kytos wrapper for Zabbix"
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-l", "--url", dest="url", help="URL for your Kytos REST API", default=url)
+parser.add_argument("-u", "--user", dest="username", help="Username to authenticate into Kytos API")
+parser.add_argument("-p", "--pass", dest="password", help="Password to authenticate into Kytos API")
+parser.add_argument("-f", "--authfile", dest="authfile", help="Authentication file containing username (first line) and password (second line) to authenticate into Kytos API")
+parser.add_argument("-c", "--cache_policy", dest="cache_policy", default=60, help="Cache policy: never, always or X seconds (default to cache for 600 seconds)")
+parser.add_argument("-o", "--monitoring_option", dest="option", type=int, default=1, choices=[1, 2, 3, 4, 5], help="Monitoring option: 1 - for monitor nodes, 2 - for monitor links, 3 - for monitor evcs (status), 4 - evc statistics, 5 - OpenFlow flows stats")
+parser.add_argument("-t", "--target", dest="target", help="Item status (0-down/others, 1-disabled, 2-up/primary, 3-up/backup). Argument is the item id to be monitored (depending on the -o option).")
+parser.add_argument("-z", "--zabbix_output", dest="count_output", type=int, choices=[1, 2], help="Zabbix LLD: (1) Count number of lines in each output or (2) list-only registers", default=None)
+parser.add_argument("-s", "--stats", dest="stats", type=int, default=1, choices=[1, 2, 3, 4], help="EVC statistics type: 1 - bytes/UNI_A, 2 - bytes/UNI_Z , 3 - packets/UNI_A, 4 - packets/UNI_Z")
+
+args = parser.parse_args()
+
+if args.option == 4 and not args.target:
+    sys.stderr.write('error: to print statistics it requires -t\n')
+    parser.print_help()
+    sys.exit(2)
+
+if args.authfile:
+    authdata = open(args.authfile).readlines()
+    args.username = authdata[0].strip()
+    args.password = authdata[1].strip()
+
+args.url = os.environ.get("KYTOS_URL", args.url)
+args.username = os.environ.get("KYTOS_USERNAME", args.username)
+args.password = os.environ.get("KYTOS_PASSWORD", args.password)
+
+def is_valid_cache(option):
+    filename = cache[option]
+    if not os.path.isfile(filename) or os.path.getsize(filename) == 0:
+        return False
+    if args.cache_policy == 'never':
+        return False
+    elif args.cache_policy == 'always':
+        return True
+    try:
+        tmout = int(args.cache_policy)
+    except:
+        tmout = 600
+    return os.stat(filename).st_mtime > time.time() - tmout
+
+def get_data(option, url):
+    if is_valid_cache(option):
+        with open(cache[option], 'r') as file:
+            return file.read()
+
+    auth = None
+    if args.username and args.password:
+        auth = (args.username, args.password)  # assume HTTP Basic
+
+    try:
+        response = requests.get(url, auth=auth)
```

Comment:
It would be great to also set a timeout and handle `requests.exceptions.Timeout` to avoid it potentially hanging https://requests.readthedocs.io/en/latest/user/quickstart/#timeouts

> Nearly all production code should use this parameter in nearly all requests. Failure to do so can cause your program to hang indefinitely:

If we're not going to use it in production shortly it's OK to leave this for later though.

Reply:
Thanks for the suggestion @viniarck. Added on dd3a133